app/assets/javascripts/arclight/collection_navigation.js

```diff
@@ -44,47 +44,125 @@
+
+    /**
+     * Resolve the response and invoke a callback for the payload
+     */
+    this.resolve = function(callback) {
+      this.request.send(callback);
+    }
+  }
+
+  /** Class for transmitting and receiving requests from the catalog endpoint */
+  function ContentRequest($el, data) {
+    this.$el = $el;
+    this.data = data;
+
+    /**
+     * Generate the response
+     * @return {ContentResponse} the response object
+     */
+    this.getResponse = function() {
+      return new ContentResponse(this);
+    }
+
+    /**
+     * @param {function} callback a function which takes the server payload as an argument
+     */
+    this.send = function(callback) {
+      var placeholder = new Placeholder();
+      this.$el.html(placeholder.render());
+      var requestData = {
+        'f[component_level_isim][]': this.data.arclight.level,
+        'f[has_online_content_ssim][]': this.data.arclight.access,
+        'f[collection_sim][]': this.data.arclight.name,
+        'f[parent_ssi][]': this.data.arclight.parent,
+        search_field: this.data.arclight.search_field,
+        view: this.data.arclight.view || 'hierarchy',
+        limit: this.data.arclight.limit || 10
+      };
+
+      var that = this;
       $.ajax({
-        url: data.arclight.path,
-        data: {
-          'f[component_level_isim][]': data.arclight.level,
-          'f[has_online_content_ssim][]': data.arclight.access,
-          'f[collection_sim][]': data.arclight.name,
-          'f[parent_ssi][]': data.arclight.parent,
-          search_field: data.arclight.search_field,
-          view: data.arclight.view || 'hierarchy'
-        }
-      }).done(function (response) {
-        var resp = $.parseHTML(response);
-        var $doc = $(resp);
-        var showDocs = $doc.find('article.document');
-        var newDocs = $doc.find('#documents');
+        url: this.data.arclight.path,
+        data: requestData
+      }).done(function(response) {
+        this.resp = $.parseHTML(response);
+        this.$doc = $(this.resp);
+        var showDocs = this.$doc.find('article.document');
+        var newDocs = this.$doc.find('#documents');
+        var paginationElements = this.$doc.find('.pagination');
 
         // Add a highlight class for the article matching the highlight id
-        if (data.arclight.highlightId) {
-          newDocs = convertDocsForContext(data.arclight.highlightId, $doc);
+        if (that.data.arclight.highlightId) {
+          // This should be removed from the global scope
+          newDocs = convertDocsForContext(that.data.arclight.highlightId, that.$doc);
+        }
+
+        // Ensure that the pagination elements are added to the AJAX-loaded
+        // content
+        if (paginationElements.length > 0) {
+          newDocs.append(paginationElements);
+
+          // Override the handlers to refresh the content
+          var paginationLinks = paginationElements.find('a')
+          paginationLinks.on('click', function(e) {
```

Comment:
I don't really like the approach which I've proposed here, but I didn't want to proceed with further refactoring attempts.  The restructuring here so far was undertaken in order to try and take a recursive approach for invoking `$.ajax` for the AJAX requests in jQuery.

Reply:
Do we want to see if the upcoming content navigation has any implication on this? Or if we implemented ES6 support that would make life easier?

Reply:
I defer to you and others on this.  `.render` calls and ES5 Classes such as `Paginator` were proposed with the hope of just preparing to broach this subject of ES6 Classes and Module exports.

Reply:
I've found the following issues which might have design implications (potentially blocking this, or requiring significant changes):
- https://github.com/projectblacklight/arclight/issues/536
- https://github.com/projectblacklight/arclight/issues/514
- https://github.com/projectblacklight/arclight/issues/540
- https://github.com/projectblacklight/arclight/issues/534
- https://github.com/projectblacklight/arclight/issues/546

If there are others which might block this (or require changes), please let me know and I can address these.  Also, if it is preferred that I make an attempt at https://github.com/projectblacklight/arclight/issues/587, I definitely understand and can make this a priority.

Reply:
FWIW, I believe we're very close to finalizing our broad design change recommendations (I'll bring up in today's planning meeting). If we get agreement on those, we'll be able to start posting specific design updates to most or all of the tickets you noted above @jrgriffiniii.

Reply:
Thank you @ggeisler, I will be certain to adjust this in response to those recommendations.  Also, please let me know if I can assist with implementing any of those.

Reply:
🆒  so sounds like we have a good working way forward here, and we can just hold for a bit until we see the updated designs?